pinot-core/src/main/java/org/apache/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -108,4 +108,48 @@ public synchronized void stop() {
       _tasksWithValidInterval.parallelStream().forEach(PeriodicTask::stop);
     }
   }
+
+  /** @return true if task with given name exists; otherwise, false. */
+  public boolean hasTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private PeriodicTask getPeriodicTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return task;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Execute specified {@link PeriodicTask} immediately. If the task is already running, wait for the running task
+   * to finish before executing the task again.
+   */
+  public void execute(String periodicTaskName, String tableName) {
+    PeriodicTask periodicTask = getPeriodicTask(periodicTaskName);
+    LOGGER.info("Immediately executing periodic task {}", periodicTaskName);
```

Comment:
A misleading log if periodicTaskName is null (as per check below), so move it at least one line down. Include table name in the log if given (or include "all tables" in the log).

A rare corner case can exist here where we rename a periodic task (or add/remove periodic tasks) in some release, and each controller has a slightly different list of periodic tasks. So, addition to returning 4xx at the API, it is good to check at the destination controllers as well.  Please add a comment to this effect so that someone reading the code may not wonder why we are checking it at the API level, and also here.

Reply:
Done.